awscli/topics/return-codes.rst

```diff
@@ -0,0 +1,35 @@
+:title: AWS CLI Return Codes
+:description: Describes the various return codes of the AWS CLI
+:category: General Topics, S3, Troubleshooting
+:related command: s3, s3 cp, s3 sync, s3 mv, s3 rm
+
+These are the following return codes returned at the end of execution
+of a CLI command:
+
+* ``0`` -- Command was successful. There were no errors thrown by either
+  the CLI or by the service the request was made to.
+
+* ``1`` -- Limited to ``s3`` commands, at least one or more s3 transfers
+  failed for the command executed.
+
+* ``2`` -- Limited to ``s3`` commands, at least one or more files marked
```

Comment:
This can also happen for bad CLI commands that don't make it out of the argparse layer.  Not sure how you'd want to word that.

```
$ aws ec2 run-instances
usage: aws [options] <command> <subcommand> [parameters]
aws: error: argument --image-id is required
$ echo $?
2
```